### PR TITLE
Populate and substitute variables on dashboard load

### LINKED ISSUE #12470: Detect, populate, and substitute variables when loading a dashboard
To load a dashboard:

1. Load all the variables the current user has access to, the dashboard, and all of the views for the dashboard
2. Scan the queries in each view to find which variables are in use on the dashboard
3. [Populate the values for the variables in use](https://github.com/influxdata/influxdb/blob/ae5bdeb997c2d8b21321f0d377a18d5ba3ca3017/ui/src/variables/actions/index.ts#L168)
4. Render the dashboard and all of its views
5. Execute the queries for each view, substituting the appropriate variable values where necessary

### PR DESCRIPTION
**Pull request by chnn**
Closes #12470

This enables a user to create a cell that uses an external variable (e.g. `from(bucket: v.my_bucket)`). Then when that dashboard is first loaded, an appropriate value for the `my_bucket` variable will be loaded, selected, and substituted before executing the query for that cell.

It doesn't yet refetch the values for variables when a cell is created or edited, and doesn't allow for selecting different values. These will follow in a future PR.

### CODE REVIEW

File: ui/src/variables/selectors/index.tsx [hunk 1]
```diff
@@ -19,19 +25,19 @@ const getVariablesForOrgMemoized = memoizeOne(
...
   state: AppState,
   contextID: string
 ): ValueSelections => {
-  const contextValues: VariableValues = get(
-    state,
-    `variables.values.${contextID}.values`,
-    {}
-  )
+  const contextValues: VariableValues =
+    get(state, `variables.values.${contextID}.values`) || {}
```
Comment (bthesorceror) [inline ui/src/variables/selectors/index.tsx]: does this sometimes have the key but it is set to null?

Comment (chnn) [reply ui/src/variables/selectors/index.tsx]: Yep

File: ui/src/variables/selectors/index.tsx [hunk 2]
```diff
@@ -48,3 +54,29 @@ export const getValueSelections = (
...
+  (
+    valuesState: ValuesState,
+    variablesState: VariablesState
+  ): VariableAssignment[] => {
+    if (!valuesState) {
+      return []
+    }
+
+    return Object.entries(valuesState.values).map(([variableID, values]) => {
+      const variableName = variablesState[variableID].variable.name
```
Comment (bthesorceror) [inline ui/src/variables/selectors/index.tsx]: what do you think about using a `get` here?

File: ui/src/shared/components/cells/Cell.tsx [hunk 3]
```diff
@@ -133,21 +112,17 @@ class CellComponent extends Component<Props> {
   }
 }
 
-const mstp = (state: AppState, ownProps: PassedProps): StateProps => {
-  const entry = state.views[ownProps.cell.id]
+const mstp = (state: AppState, ownProps: OwnProps): StateProps => {
+  const entry = state.views.views[ownProps.cell.id]
```
Comment (bthesorceror) [inline ui/src/shared/components/cells/Cell.tsx]: naming is weird here `views.views`, but more relevant is maybe we use a `get` here?

Comment (chnn) [reply ui/src/shared/components/cells/Cell.tsx]: I extracted it to a selector function, and used `get` within the selector.

The naming is weird... it comes up any time we have a slice of Redux state that has loading state:

```
interface ViewsState {
  status: RemoteDataState
  views: {
    [viewID: string]: {
      status: RemoteDataState
      view: View
    }
  }
}
```

This also happens with some variation in the variables, telegrafs, buckets, and tokens state. Consistency would be nice eventually ¯\_(ツ)_/¯

File: ui/src/variables/selectors/index.tsx (hunk 2 above)
Comment (chnn) [reply ui/src/variables/selectors/index.tsx]: 👍

**Pull request merged**